server.py

```diff
@@ -141,13 +170,13 @@ def merge(response):
     try:
         db.session.merge(response)
         db.session.commit()
-    except SQLAlchemyError as e:
-        logger.info("Unable to save response", error=e)
-        raise SQLAlchemyError
     except IntegrityError as e:
         logger.info("Integrity error in database. Rolling back commit",
                     error=e)
-        raise IntegrityError
+        raise
+    except SQLAlchemyError as e:
+        logger.info("Unable to save response", error=e)
+        raise SQLAlchemyError
```

Comment:
The log statement should be error level

Also the raise call should change to either
raise e
or
raise SQLAlchemyError()

The first one is probably better